File: OffByOne/src/OffByOne.Ql/Ast/Statements/Statement.cs
namespace OffByOne.Ql.Ast.Statements
{
    using OffByOne.Ql.Visitors.Contracts;

    public abstract class Statement : AstNode, IVisitableStatement
    {
        public abstract TResult Accept<TResult, TContext>(IStatementVisitor<TResult, TContext> visitor, TContext context)
            where TContext : IContext;
    }
}

File: TBD/QL/src/ql/ast/Statement.java
package ql.ast;

import ql.ast.visistor.interfaces.BaseVisitor;

/**
 * Created by Erik on 6-2-2017.
 */
public abstract class Statement extends ASTNode {

    public Statement(int rowNumber) {
        super(rowNumber);
    }

    public abstract  <T> T accept(BaseVisitor<T> visitor);
}
